src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -133,7 +468,88 @@ instance (Arithmetic a, KnownNat n) => BoolType (ByteString n (ArithmeticCircuit
                 flipped <- forM bits $ \b -> newAssigned (\p -> one - p b)
                 horner flipped
 
-    l || r = binOp l r (\i j x -> x i + x j - x i * x j)
+    l || r = bitwiseOperation l r (\i j x -> x i + x j - x i * x j)
+
+    l && r = bitwiseOperation l r (\i j x -> x i * x j)
+
+    xor l r = bitwiseOperation l r (\i j x -> x i + x j - (x i * x j + x i * x j))
+
 
-    l && r = binOp l r (\i j x -> x i * x j)
+instance
+  ( KnownNat wordSize
+  , KnownNat n
+  , wordSize <= n
+  , 1 <= wordSize
+  , 1 <= n
+  , Mod n wordSize ~ 0
```

Comment:
Same here: `Mod n wordSize ~ 0` logically implies `1 <= wordSize`. And, together with `1 <= n` -- that `wordSize <= n`